controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -208,6 +210,23 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				},
 			),
 		).
+		// Watcher for storageClass required to update values related to replica-1
+		// in ocs-operator-config configmap, if storageClass changes
+		Watches(
+			&storagev1.StorageClass{},
+			handler.EnqueueRequestsFromMapFunc(
+				func(context context.Context, obj client.Object) []reconcile.Request {
+					// Only reconcile if the storageClass is external & has topologyConstrainedPools set
+					sc := obj.(*storagev1.StorageClass)
+					if strings.Contains(sc.Name, "external") && sc.Parameters["topologyConstrainedPools"] != "" {
```

Comment:
Can we pls remove the first condition? There might be customers who do not have an external name in the storageclasses. They may have created the storage cluster from the CLI.